### ISSUE TITLE
Add ignore: ["empty-lines"] option to no-eol-whitespace

### ISSUE THREAD

**Issue by hlomzik** [stylelint/stylelint#1273]
> Describe the feature request (new option)

The rule `no-eol-whitespace` checks for trailing whitespaces, which is very good. But it complains even about "blank" lines made of whitespaces only. There are some code styles which allow (or request) such behaviour (and some tools have settings for this): some questions about this ([1](http://stackoverflow.com/questions/15051201/indent-blank-lines-intellij-idea), [2](http://stackoverflow.com/questions/5917956/why-is-indentation-in-empty-lines-bad)); [`skipBlankLines` option for `no-trailing-spaces` rule for ESLint](http://eslint.org/docs/rules/no-trailing-spaces); [`ignoreEmptyLine` option for `disallowTrailingWhitespace` rule for JSCS](http://jscs.info/rule/disallowTrailingWhitespace); [`Keep indents on empty lines` setting for WebStorm](http://imgur.com/vVVlIBM).

Though this behaviour could be controversial, I think it's good to have a way to allow it by adding option like `ignore: "blank-lines"` to `no-eol-whitespace` rule.

> What CSS is needed to reproduce this issue?

```css
.container {
••position: relative;
••
••font-weight: bold;
}
```

`•` represents whitespaces; note the 3rd line which has "trailing" spaces at the same indent as a whole block. Now you'll have error from stylelint: `21:4  ✖  Unexpected whitespace at end of line   no-eol-whitespace`. 

> What did you expect to happen?

 With option enabled: No warnings for lines made of whitespaces only:

```json
{ "rules": {
  "no-eol-whitespace": [ true, {
    "ignore": "blank-lines"
  } ]
} }
```

**Comment by jeddy3** [comment]
> I think it's good to have a way to allow it by adding option like ignore: "blank-lines" to no-eol-whitespace rule.


Sounds good to me.

**Comment by davidtheclark** [comment]
No, let's call it `empty-lines`, though, instead of `blank-lines`.

**Comment by jeddy3** [comment]
@hlomzik Thanks for the suggestion and clear write-up. Feel free to contribute this option. There's [a section](https://github.com/stylelint/stylelint/blob/master/docs/developer-guide/rules.md#adding-options-to-existing-rules) in the Developer Guide on how to get started.
